test/system/oauth2_test.rb

```diff
@@ -2,9 +2,13 @@
 
 class Oauth2Test < ApplicationSystemTestCase
   def test_authorized_applications
-    sign_in_as(create(:user))
+    sign_in_as(create(:user)) # successful login redirects to a page that authorizes the website app
     visit oauth_authorized_applications_path
 
+    accept_alert do
+      click_link "Revoke Access"
+    end
```

Comment:
Rather than adding this (and the comment above) all of which took me a few minutes to understand, wouldn't it be simpler just to change the assert below to expect an application?

This test is just trying to check that the authorized applications page exists and displays rather than the specific contents...

An alternative which would be clearer would be to drop the comment and add an assertion above the revoke to check that the website application is listed and then revoke it and check you get the no apps message?

Reply:
> Rather than adding this (and the comment above) all of which took me a few minutes to understand, wouldn't it be simpler just to change the assert below to expect an application?
> This test is just trying to check that the authorized applications page exists and displays rather than the specific contents...

I was asked to test _the specific contents_ of no registered apps: https://github.com/openstreetmap/openstreetmap-website/pull/4434#issuecomment-1869571084. I'm not testing for the website app to be listed here. I'm adapting the already existing test of the message that had a missing translation up until recently to work with the auto-registering website app, which you asked to register for every system test.

> An alternative which would be clearer would be to drop the comment and add an assertion above the revoke to check that the website application is listed and then revoke it and check you get the no apps message?

What is the alternative? I'm revoking the access and checking the message here. Do you want me to revoke it by manipulating models/db then reload the page to check the message?

Reply:
The alternative I was suggesting was to remove the comment after the `create` and replace it with an assertion before the revoke that check there is a single app listed and that it's the web site app - it serves the same purpose as the comment in explaining what the revoke is doing but I think it's clearer and also extends testing of the page to test that it lists authorized apps correctly.

Reply:
I replaced the comment with the assert, but this is really testing side effects of `sign_in_as()`. You could have another login method that avoids visiting the pages that auto-grant permissions to the website app.